presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -466,6 +466,19 @@ private void destroyIfNecessary()
         return blocked;
     }
 
+    private static boolean isBlocked(Operator operator)
```

Comment:
This is good idea, but is not compatible with existing operators. Cancelling what they return from `isBlocked` is going to break them, so revising operator implementations needs to be done first.

I've, however, change the implementation so that it doesn't create any more futures than it used to create.
Please consider merging this as it is now